sky/shell/gpu/direct/rasterizer_direct.cc

```diff
@@ -4,25 +4,33 @@
 
 #include "flutter/sky/shell/gpu/direct/rasterizer_direct.h"
 
+#include <string>
+#include <utility>
+
+#include "flutter/common/threads.h"
 #include "flutter/glue/trace_event.h"
-#include "mojo/public/cpp/system/data_pipe.h"
 #include "flutter/sky/engine/wtf/PassRefPtr.h"
 #include "flutter/sky/engine/wtf/RefPtr.h"
 #include "flutter/sky/shell/gpu/picture_serializer.h"
 #include "flutter/sky/shell/platform_view.h"
 #include "flutter/sky/shell/shell.h"
+#include "mojo/public/cpp/system/data_pipe.h"
 #include "third_party/skia/include/core/SkCanvas.h"
 #include "third_party/skia/include/core/SkPicture.h"
 
 namespace sky {
 namespace shell {
 
 RasterizerDirect::RasterizerDirect()
-    : binding_(this), platform_view_(nullptr), weak_factory_(this) {}
+    : platform_view_(nullptr), weak_factory_(this) {
+  auto weak_ptr = weak_factory_.GetWeakPtr();
+  blink::Threads::Gpu()->PostTask(
+      [weak_ptr]() { Shell::Shared().AddRasterizer(weak_ptr); });
+}
 
 RasterizerDirect::~RasterizerDirect() {
   weak_factory_.InvalidateWeakPtrs();
-  Shell::Shared().PurgeRasterizers();
+  blink::Threads::Gpu()->PostTask([]() { Shell::Shared().PurgeRasterizers(); });
```

Comment:
It doesn't get created on the GPU thread but it does get destroyed on the GPU thread.  Having this code here is confusing and makes this issue less clear for future people editing this file.